lib/vagrant/util/network_ip.rb

```diff
@@ -3,11 +3,23 @@
 module Vagrant
   module Util
     module NetworkIP
+
+      DEFAULT_MASK = "255.255.255.0".freeze
+
+      def initialize
+        @logger = Log4r::Logger.new("vagrant::util::NetworkIP")
```

Comment:
Quick question about this....it looks like the way this is used is by calling `include` on this class and invoking the class method `network_address` directly rather than on a new `NetworkIP` object. I'm not 100% sure about this, but would that bypass the initialize here and make `@logger` nil below?

Reply:
Manually testing it out seems to work. I also updated the tests so the logger bit gets run and it seems to work. That being said, when I included this module using `extend` instead of `include` then `@logger` will be `nil`. I think it has to do with how Ruby inserts classes into the object hierarchy when using `include` vs `extend`. 
I think if `network_address` is meant to be a static method then this will break? Maybe I'll chuck this in `network_address`?

Reply:
Ok! That's what I wasn't sure about, the extend versus include. I always confuse the two! I'd say it doesn't hurt to move it into the `network_address` method to be safe, in case a caller ends up `extend`ing the library instead of `include`.

Reply:
Also, in the past for classes like this we've ended up defining a "global" logger here. Not ideal, but then we could define the logger once at the top to be used for any method in the class. Something like this at the top:
```ruby
LOGGER = Log4r::Logger.new(...)
```

Then we can use `LOGGER` in the class without defining it multiple times :thinking: